### PR TITLE
MWPW-153245 [catalog] analytics fixes

### PR DESCRIPTION
**Pull request by npeltier**
- listen to newly created WC events and call analytics API directly (see https://github.com/orgs/adobecom/discussions/2512),
- only consume one (minified) sidenav dependency,
- decorate catalog instead of collection (less levels),
- decorate card name at the link level, to prevent having card information at the end of the event,

Resolves: [MWPW-136881](https://jira.corp.adobe.com/browse/MWPW-136881)
milo PR: https://github.com/adobecom/milo/pull/2610
mas PR: https://github.com/adobecom/mas/pull/42

**Test URLs:**
- Before: https://main--cc--adobecom.hlx.live/?martech=off
- After: https://mwpw-153245--cc--npeltier.hlx.live/?martech=off

### CODE REVIEW

File: creativecloud/blocks/catalog/catalog.js [hunk 1]
```diff
@@ -6,6 +6,57 @@ const { loadStyle } = await import(`${miloLibs}/utils/utils.js`);
...
+  }
+  // eslint-disable-next-line no-underscore-dangle
+  window._satellite?.track('event', {
+    xdm: {},
+    data: { web: { webInteraction: { name: `${eventName}|${daaLhValue}` } } },
+  });
+}
+
+function enableAnalytics(catalog, merchCards, sidenav) {
+  merchCards.querySelectorAll('merch-card').forEach(async (card) => {
```
Comment (mirafedas) [inline creativecloud/blocks/catalog/catalog.js:28]: Is `async` needed here?

File: creativecloud/blocks/catalog/catalog.js [hunk 2]
```diff
@@ -6,6 +6,57 @@ const { loadStyle } = await import(`${miloLibs}/utils/utils.js`);
 // Helps with TBT: MWPW-145127
 loadStyle(`${miloLibs}/blocks/global-navigation/features/profile/dropdown.css`);
 
+function handleCustomAnalyticsEvent(eventName, element) {
```
Comment (mirafedas) [inline creativecloud/blocks/catalog/catalog.js:9]: Should we maybe add some tests for `handleCustomAnalyticsEvent` and `enableAnalytics`?

**Review by mirafedas — CHANGES_REQUESTED** [review]

**Comment by npeltier** [comment]
@mirafedas 
image

**Review by mirafedas — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]

**Comment by aishwaryamathuria** [comment]
@npeltier @mirafedas @3ch023 Could you confirm if PR is verified by adding the 'verified' label?

**Comment by Roycethan** [comment]
> @npeltier @mirafedas @3ch023 Could you confirm if PR is verified by adding the 'verified' label?

Done added label , although in Milo process we will give importance to Ready for stage label, if its there that means its verified. Thanks

**Review by aishwaryamathuria — APPROVED** [review]

**Pull request merged**